scala/scala_cross_version.bzl

```diff
@@ -110,24 +101,38 @@ _declare_scalac_provider(
 _generate_scalac_build_file = repository_rule(
     implementation = _generate_scalac_build_file_impl,
     attrs = {
-        "archive": attr.string(),
-        "version_underscore": attr.string()
+        "version_underscore": attr.string(),
     })
 
-def new_scala_repository(name, version):
-  major_version = version[:version.find(".", 2)]
-  archive = "{name}_imports".format(name = name)
-  native.new_http_archive(
-      name = archive,
-      strip_prefix = "scala-{version}".format(version = version),
-      url =
-      "https://downloads.lightbend.com/scala/{version}/scala-{version}.tgz".
-      format(version = version),
-      build_file_content = _generate_scala_imports(major_version),
+def new_scala_repository(name, version, maven_servers):
+  version_underscore = version.replace(".", "_")
+
+  _scala_maven_import_external(
+      name = "io_bazel_rules_scala_scala_library_{}".format(version_underscore),
```

Comment:
I lost a bit of context but why do we need to declare these twice as external dependencies. Once in scala.bzl and once here.
Also if there's a good enough reason then maybe we need the dict sha trick here as well

Reply:
In `scala.bzl` we only import scalatest, scala_xml and parser-combinators (which are only specific to major scala versions) and here we import everything that's specific to minor scala versions.

These imports could be in `scala.bzl` as well but if we want to be able to have multiple versions in the future they need to be here.